confmap/provider.go

```diff
@@ -100,21 +102,33 @@ func WithRetrievedClose(closeFunc CloseFunc) RetrievedOption {
 	}
 }
 
-// NewRetrievedFromMap returns a new Retrieved instance that contains a Map data.
-// * conf the Map that will be merged to the given map in the MergeTo.
-// * CloseFunc specifies a function to be invoked when the configuration for which it was
-//   used to retrieve values is no longer in use and should close and release any watchers
-//	 that it may have created.
-func NewRetrievedFromMap(conf *Conf, opts ...RetrievedOption) Retrieved {
+// NewRetrieved returns a new Retrieved instance that contains the data from the raw deserialized config.
+func NewRetrieved(rawConf map[string]interface{}, opts ...RetrievedOption) (Retrieved, error) {
 	set := retrievedSettings{}
 	for _, opt := range opts {
 		opt(&set)
 	}
-	return Retrieved{conf: conf, closeFunc: set.closeFunc}
+	return Retrieved{conf: NewFromStringMap(rawConf), closeFunc: set.closeFunc}, nil
+}
+
+// NewRetrievedFromYAML returns a new Retrieved instance that contains the deserialized data from the yaml bytes.
+// * yamlBytes the yaml bytes that will be deserialized.
+// * opts specifies options associated with this Retrieved value, such as CloseFunc.
+func NewRetrievedFromYAML(yamlBytes []byte, opts ...RetrievedOption) (Retrieved, error) {
```

Comment:
Could this be an internal function on `confmap/provider/internal`? The only use outside of the provider folder is on `service/mapresolver_test.go`, would it be feasible to remove it from there? I think `NewRetrieved` is general enough for fulfilling any use case, so it would be nice to have just that.

Reply:
I can do that for the moment until outside world will file an issue that they need it as well :)

Reply:
Actually I like it, done.